### PR TITLE
(SIMP-2234) Create resolv module

### PR DESCRIPTION
**Pull request by jeefberkey**
Migrated content from simplib::resolv

SIMP-2234 #close
SIMP-1679 #comment Created resolv module

### CODE REVIEW

File: spec/acceptance/nodesets/centos-6-x64.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+HOSTS:
```
Comment (jeefberkey) [inline spec/acceptance/nodesets/centos-6-x64.yml]: Trim nodesets

File: spec/acceptance/suites/default/class_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+require 'spec_helper_acceptance'
```
Comment (jeefberkey) [inline spec/acceptance/suites/default/class_spec.rb]: Remove generated code

**Comment by nick-markowski** [comment]
dep lords, pls.

File: spec/classes/init_spec.rb [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        context 'node_is_nameserver' do
+          let(:facts) { facts.merge({:ipaddress => '10.0.2.15'}) }
+
+          let(:params) {{
+            :servers => ['1.2.3.4','5.6.7.8','10.0.2.15']
+          }}
+
+          it { is_expected.to compile.with_all_deps }
+          it { is_expected.not_to contain_class('named::caching') }
+          # it { is_expected.to contain_class('named') }
```
Comment (nick-markowski) [inline spec/classes/init_spec.rb]: This is a valid test, uncomment it

File: manifests/init.pp [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+#   true => If the server where puppet is being run is in the list of
+#           $servers then automatically configure named.
+#   false => Do not autoconfigure named.
+# @param caching
+#   *If* the $servers array above starts with '127.0.0.1' or '::1', then
+#   the system will set itself up as a caching nameserver unless this is set
+#   to false.
+#
+class resolv (
+  Array           $servers        = simplib::lookup('simp_options::dns::servers', { 'default_value' => ['127.0.0.1'], 'value_type'      => Array[String] }),
```
Comment (nick-markowski) [inline manifests/init.pp]: This (and below) should probably be validated as arrays of hostnames

**Pull request merged**